plugins/module_utils/module_helper.py

```diff
@@ -296,17 +296,25 @@ def find_format(_param):
         param_list = params if params else self.module.params.keys()
 
         for param in param_list:
-            if param in self.module.argument_spec:
-                if param not in self.module.params:
-                    continue
-                fmt = find_format(param)
-                value = self.module.params[param]
+            if isinstance(param, dict):
+                if len(param.keys()) != 1:
+                    raise ModuleHelperException("run_command parameter as a dict must "
+                                                "contain only one key: {0}".format(param))
+                _param = list(param.keys())[0]
+                fmt = find_format(_param)
+                value = param[_param]
+            elif isinstance(param, str):
+                if param in self.module.argument_spec:
+                    fmt = find_format(param)
+                    value = self.module.params[param]
+                elif param in extra_params:
+                    fmt = find_format(param)
+                    value = extra_params[param]
+                else:
+                    raise ModuleHelperException("Cannot determine value for parameter: {0}".format(param))
```

Comment:
The old code did the following:
```suggestion
                    continue
```

Reply:
That one actually changes the behaviour, but I believe that's a good one: right now if it cannot determine the value for that parameter it will silently ignore that param, which might be confusing to debug.

Reply:
That's true, but it could change behavior of existing code. How about emitting a deprecation warning instead, saying that this will result in an exception from c.g 4.0.0 on?

Reply:
that's more a warning than a deprecation isn't it?

Reply:
It's both, since eventually it will switch from a warning to an error. So we add a warning and deprecate the current behavior.